cli/cmd/doc.go

```diff
@@ -0,0 +1,43 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+)
+
+var outputPath string
+
+func newCmdDoc() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:    "doc",
+		Hidden: true,
+		Short:  "Generate YAML documentation for the linkerd executable (hidden command)",
+		Args:   cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return generateDocs(outputPath)
+		},
+	}
+
+	cmd.Flags().StringVarP(&outputPath, "output", "o", "", "Output path for the resulting YAML files")
+
+	return cmd
+}
+
+func generateDocs(outputPath string) error {
+	if outputPath == "" {
+		return errors.New("you must specify an output path for the resulting YAML files using the -o or --output flag")
+	}
+
+	fmt.Println("Generating YAML docs...")
+
+	err := doc.GenYamlTree(RootCmd, outputPath)
```

Comment:
thinking about this a bit more, the cli never writes to disk, i'd like to preserve that pattern. also, since https://github.com/linkerd/website/blob/master/linkerd.io/data/cli.yaml is a single file, i think it's fine to output them all together, perhaps with `---` separators.

Reply:
@siggy As you can see [here](https://github.com/spf13/cobra/blob/master/doc/yaml_docs.md), Cobra generates one YAML file per command, and I don't believe that there's a way to combine them. I don't really see a way around this, unfortunately.

Reply:
@lucperkins It looks like there's some more flexiblity with the `GenYaml` and `GenYamlCustom` commands:
https://github.com/spf13/cobra/blob/master/doc/yaml_docs.md#generate-yaml-docs-for-a-single-command

Reply:
My goal is to substantially improve the Linkerd documentation in a short period of time. I apologize if this comes across as undiplomatic, but I think that this lies outside of my mandate.